### DISCUSSION TITLE
SolidJS URQL Binding

### DISCUSSION THREAD

**Discussion by captDaylight** [urql-graphql/urql#3242] [Ideas]
I'm interested in starting a SolidJS URQL binding into the fold. @tqwewe has done work on a [project](https://github.com/tqwewe/solid-urql), but it looks like it hasn't been contributed to since February 2022. 

Is there interest in incorporating this into the repo?

**Comment by kitten** [comment]
We actually have an old PR for this, however, I didn't get around to reviewing the code and refactoring it, i.e. checking whether it's idiomatic and working well. And the big blocker here was figuring out some testing (not high-fidelity unit tests, but at least as much as `@urql/vue` has) https://github.com/urql-graphql/urql/pull/2527

So, I'm still hoping to revive that, but more work will definiteiyl be needed to get there still and I haven't found the time to pick it up 😅

**Reply by captDaylight** [reply]
Cool, thanks for pointing to the branch! Having a look through it now

**Comment by ofarukaydin** [comment]
I've written binding for the project I'm working on myself. It works with suspense/solid start/streaming and deferStream. Heavily inspired by tanstack-query solid bindings. You can play with it while waiting for official support.  It also supports solids hydration with ssrExchange. You need to request __typename for every field explicitly for hydration to work.
Here is sandbox for the code:
https://stackblitz.com/edit/github-xzthyv?file=src%2Froutes%2Findex.tsx
